src/PuliPluginImpl.php

```diff
@@ -141,9 +150,39 @@ public function postAutoloadDump()
 
         $factoryFile = Path::makeAbsolute($factoryFile, $this->rootDir);
 
-        $this->insertFactoryClassConstant($autoloadFile, $factoryClass);
-        $this->insertFactoryClassMap($classMapFile, $vendorDir, $factoryClass, $factoryFile);
-        $this->setBootstrapFile($autoloadFile);
+        $autoload = $this->composer->getPackage()->getAutoload();
+        $autoload['classmap'][] = $factoryFile;
+        $this->composer->getPackage()->setAutoload($autoload);
+
+        if (!file_exists($factoryFile)) {
+            $filesystem = new Filesystem();
+            $filesystem->dumpFile($factoryFile, '<?php class '.$factoryClass.' {}');
```

Comment:
Yeah, Composer will only find classes that exist. It will be overwritten by Puli later on; will add a note.